lib/Ix/DBIC/Result.pm

```diff
@@ -274,43 +470,250 @@ sub ix_finalize ($class) {
+
+=cut
 
 sub ix_create_error  { return; } # ($self, $ctx, \%error)
 sub ix_update_error  { return; } # ($self, $ctx, \%error)
 
+=method ix_created($ctx, $row)
+
+A hook method called inside the Foo/set#create transaction.
+
+=method ix_updated($ctx, $row, \%changes)
+
+A hook method called inside the Foo/set#update transaction. C<$changes> is a
+hashref with both C<old> and C<new> keys, so hooks can inspect the changes and
+take action only in certain cases.
+
+=method ix_destroyed($ctx, $row)
+
+A hook method called inside the Foo/set#destroy transaction.
+
+=cut
+
 sub ix_created   { } # ($self, $ctx, $row)
 sub ix_destroyed { } # ($self, $ctx, $row)
 
 # The input to ix_updated is not trivial to compute, so it is only called if
 # present, so we don't define it in the base class. -- rjbs, 2017-01-06
 # sub ix_updated   { } # ($self, $ctx, $row, \%changes)
 
+=method ix_postprocess_create($self, $ctx, \@rows)
+
+A postprocess hook called after Foo/set#create operations.
+
+=method ix_postprocess_update($self, $ctx, \%updated)
+
+A postprocess hook called after Foo/set#create operations.
+
+=method ix_postprocess_destroy($self, $ctx, \@row_ids)
+
+A postprocess hook called after Foo/set#create operations.
+
+=cut
+
 sub ix_postprocess_create  { } # ($self, $ctx, \@rows)
 sub ix_postprocess_update  { } # ($self, $ctx, \%updated)
 sub ix_postprocess_destroy { } # ($self, $ctx, \@row_ids)
 
+# I am not going to document this publicly, but its signature is
+# ($self, $ctx, $get_arg, $results)
 sub _return_ix_get   { return $_[3]->@* }
 
+# Methods without base implementations below
+
+=method ix_query_filter_map
+
+Required for query-enabled rclasses. This should return a hashref that defines
+how your rclass behaves when filtering. Its keys are properties that are valid
+for filtering, and its values are hashrefs defining the filter.
+
+Say your method returned the following hashref:
+
+    {
+      batch => { required => 1 },
+      types => {
+        cond_builder => sub ($types) {
+          return { type => $types },
+        },
+        differ => sub ($entity, $filter) {
+          # This difffers if its type is not in filter list
+          my $found = first { ! differ($entity->type, $_) } @$filter;
+
+          return $found ? 0 : 1;
+        },
+      },
+    }
+
+This means that your rclass would allow the keys 'batch' and 'types' for
+filtering. You can provide a C<required> key to require a particular filter. If
+C<cond_builder> is provided, its return value is used to pass into a DBIC
+C<search> argument. If you don't provide C<cond_builder> (as we do here with
+'batch'), the search will use simple string equality.
+
+Finally, you can provide a C<differ> key that controls the way elements are
+compared in ix_query_changes. If you don't provide one, C<differ()> from
+L<Ix::Util> is used. If you do provide one, it should be a coderef, which is
+passed the database entity and the relevant argument from the client-provided
+filter
```

Comment:
maybe specify what the differ should return

Reply:
Good call.